modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/Slot.java

```diff
@@ -52,6 +61,16 @@
     private boolean isSlotActive;
 
     /**
+     * Indicates whether the slot is a fresh one or an overlapped one
+     */
+    private boolean isAnOverlappingSlot;
```

Comment:
Once overlapped we keep this boolean true till it gets deleted? 
For instance when we change state from say, overlapped -> assigned we still need to check its ovelappin. Is that why we keep this information separate rather than using SlotState for this purpose?